tests/bluetooth/tester/src/audio/btp_bap_audio_stream.c

```diff
@@ -2,188 +2,235 @@
-				stream->last_req_seq_num);
-
-			err = bt_bap_stream_send(bap_stream, buf, 0);
-			if (err != 0) {
-				LOG_ERR("Failed to send audio data to stream %p, err %d",
-					bap_stream, err);
-				net_buf_unref(buf);
-			}
-		}
+int btp_bap_audio_stream_tx_register(struct btp_bap_audio_stream *stream)
+{
+	if (stream == NULL) {
+		return -EINVAL;
+	}
 
-		/* Free ring buffer memory */
-		err = ring_buf_get_finish(&audio_ring_buf, size);
-		if (err != 0) {
-			LOG_ERR("Error freeing ring buffer memory: %d", err);
+	if (!btp_bap_audio_stream_can_send(stream)) {
+		return -EINVAL;
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(tx_streams); i++) {
+		if (tx_streams[i].bap_stream == NULL) {
+			tx_streams[i].bap_stream = audio_stream_to_bap_stream(stream);
+
+			LOG_INF("Registered %p (%p) for TX", stream, tx_streams[i].bap_stream);
+
+			return 0;
 		}
 	}
 
-	k_work_schedule_for_queue(&iso_data_work_q, dwork,
-				  K_USEC(bap_stream->qos->interval));
+	return -ENOMEM;
 }
 
-void btp_bap_audio_stream_started(struct btp_bap_audio_stream *a_stream)
+int btp_bap_audio_stream_tx_unregister(struct btp_bap_audio_stream *stream)
 {
-	struct bt_bap_ep_info info;
-	struct bt_bap_stream *bap_stream = audio_stream_to_bap_stream(a_stream);
+	const struct bt_bap_stream *bap_stream;
 
-	/* Callback called on transition to Streaming state */
+	if (stream == NULL) {
+		return -EINVAL;
+	}
 
-	LOG_DBG("Started stream %p", bap_stream);
+	bap_stream = audio_stream_to_bap_stream(stream);
 
-	(void)bt_bap_ep_get_info(bap_stream->ep, &info);
-	if (info.can_send == true) {
-		/* Schedule first TX ISO data at seq_num 1 instead of 0 to ensure
-		 * we are in sync with the controller at start of streaming.
-		 */
-		a_stream->seq_num = 1;
+	for (size_t i = 0U; i < ARRAY_SIZE(tx_streams); i++) {
+		if (tx_streams[i].bap_stream == bap_stream) {
+			memset(&tx_streams[i], 0, sizeof(tx_streams[i]));
 
-		/* Run audio clock work in system work queue */
-		k_work_init_delayable(&a_stream->audio_clock_work, audio_clock_timeout);
-		k_work_schedule(&a_stream->audio_clock_work, K_NO_WAIT);
+			LOG_INF("Unregistered %p for TX", bap_stream);
 
-		/* Run audio send work in user defined work queue */
-		k_work_init_delayable(&a_stream->audio_send_work, audio_send_timeout);
-		k_work_schedule_for_queue(&iso_data_work_q, &a_stream->audio_send_work,
-					  K_USEC(bap_stream->qos->interval));
+			return 0;
+		}
 	}
-}
 
-void btp_bap_audio_stream_stopped(struct btp_bap_audio_stream *a_stream)
-{
-	/* Stop send timer */
-	k_work_cancel_delayable(&a_stream->audio_clock_work);
-	k_work_cancel_delayable(&a_stream->audio_send_work);
+	return -ENODATA;
 }
 
-uint8_t btp_bap_audio_stream_send_data(const uint8_t *data, uint8_t data_len)
+void btp_bap_audio_stream_tx_init(void)
 {
-	return ring_buf_put(&audio_ring_buf, data, data_len);
+	static bool thread_started;
+
+	if (!thread_started) {
+		static K_KERNEL_STACK_DEFINE(tx_thread_stack, 1024U);
```

Comment:
Since this is defined at the application layer, there's not much point in introducing a Kconfig; there are no users above this, and the stack requirement won't change based on any configuration.